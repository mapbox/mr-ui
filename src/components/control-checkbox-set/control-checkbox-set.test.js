import React from 'react';
import { shallow } from 'enzyme';
import { testCases } from './control-checkbox-set-test-cases';

describe('ControlCheckbox', () => {
  let testCase;
  let wrapper;

  describe(testCases.basic.description, () => {
    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('onChange gets called with input id and value', () => {
      const mockEventOne = {
        target: {
          value: 'foo'
        }
      };
      const mockEventTwo = {
        target: {
          value: 'foobar'
        }
      };
      wrapper.find('input').first().props().onChange(mockEventOne);
      wrapper.find('input').first().props().onChange(mockEventTwo);
      wrapper.find('input').first().props().onChange(mockEventOne);
      wrapper.update();
      expect(wrapper).toMatchSnapshot();
      expect(testCase.props.onChange).toHaveBeenCalledTimes(3);
      expect(testCase.props.onChange).toHaveBeenCalledWith(
        ['foobar'],
        'testinput'
      );
    });
  });

  describe(testCases.disabled.description, () => {
    beforeEach(() => {
      testCase = testCases.disabled;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('extra props works', () => {
      const props = wrapper.find('input').last().props();
      expect(props).toHaveProperty('disabled', true);
    });
  });

  describe(testCases.allOptions.description, () => {
    beforeEach(() => {
      testCase = testCases.allOptions;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
