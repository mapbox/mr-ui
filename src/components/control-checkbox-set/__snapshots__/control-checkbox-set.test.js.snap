// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlCheckbox all options renders 1`] = `
<ControlWrapper
  id="testinput"
  themeControlWrapper="bg-red-light"
  validationError="oh no!"
>
  <div
    aria-invalid={true}
    data-test="testinput-input"
    id="testinput"
  >
    <label
      className="checkbox-container w-full"
      key="Option value 1"
    >
      <input
        aria-checked={true}
        autoFocus={true}
        checked={true}
        name="testinput"
        onChange={[Function]}
        role="checkbox"
        type="checkbox"
        value="Option value 1"
      />
      <div
        className="checkbox--red mr6 checkbox"
      >
        <Icon
          name="check"
        />
      </div>
      <span>
        Option 1
      </span>
    </label>
    <label
      className="checkbox-container w-full"
      key="Option value 2"
    >
      <input
        aria-checked={true}
        autoFocus={true}
        checked={true}
        name="testinput"
        onChange={[Function]}
        role="checkbox"
        type="checkbox"
        value="Option value 2"
      />
      <div
        className="checkbox--red mr6 checkbox"
      >
        <Icon
          name="check"
        />
      </div>
      <span>
        Option 2
      </span>
    </label>
    <label
      className="checkbox-container w-full"
      key="Option value 3"
    >
      <input
        aria-checked={false}
        autoFocus={false}
        checked={false}
        name="testinput"
        onChange={[Function]}
        role="checkbox"
        type="checkbox"
        value="Option value 3"
      />
      <div
        className="checkbox--red mr6 checkbox"
      >
        <Icon
          name="check"
        />
      </div>
      <span>
        Option 3
      </span>
    </label>
  </div>
</ControlWrapper>
`;

exports[`ControlCheckbox basic onChange gets called with input id and value 1`] = `
<ControlWrapper
  id="testinput"
>
  <div
    data-test="testinput-input"
    id="testinput"
  >
    <label
      className="checkbox-container txt-s block mb6 flex"
      key="Option value 1"
    >
      <input
        aria-checked={false}
        autoFocus={false}
        checked={false}
        name="testinput"
        onChange={[Function]}
        role="checkbox"
        type="checkbox"
        value="Option value 1"
      />
      <div
        className="mr6 inline-block checkbox--blue checkbox--s-label checkbox"
      >
        <Icon
          name="check"
        />
      </div>
      Option 1
    </label>
    <label
      className="checkbox-container txt-s block mb6 flex"
      key="Option value 2"
    >
      <input
        aria-checked={false}
        autoFocus={false}
        checked={false}
        name="testinput"
        onChange={[Function]}
        role="checkbox"
        type="checkbox"
        value="Option value 2"
      />
      <div
        className="mr6 inline-block checkbox--blue checkbox--s-label checkbox"
      >
        <Icon
          name="check"
        />
      </div>
      Option 2
    </label>
  </div>
</ControlWrapper>
`;

exports[`ControlCheckbox basic renders 1`] = `
<ControlWrapper
  id="testinput"
>
  <div
    data-test="testinput-input"
    id="testinput"
  >
    <label
      className="checkbox-container txt-s block mb6 flex"
      key="Option value 1"
    >
      <input
        aria-checked={false}
        autoFocus={false}
        checked={false}
        name="testinput"
        onChange={[Function]}
        role="checkbox"
        type="checkbox"
        value="Option value 1"
      />
      <div
        className="mr6 inline-block checkbox--blue checkbox--s-label checkbox"
      >
        <Icon
          name="check"
        />
      </div>
      Option 1
    </label>
    <label
      className="checkbox-container txt-s block mb6 flex"
      key="Option value 2"
    >
      <input
        aria-checked={false}
        autoFocus={false}
        checked={false}
        name="testinput"
        onChange={[Function]}
        role="checkbox"
        type="checkbox"
        value="Option value 2"
      />
      <div
        className="mr6 inline-block checkbox--blue checkbox--s-label checkbox"
      >
        <Icon
          name="check"
        />
      </div>
      Option 2
    </label>
  </div>
</ControlWrapper>
`;

exports[`ControlCheckbox disabled renders 1`] = `
<ControlWrapper
  id="testinput"
>
  <div
    data-test="testinput-input"
    id="testinput"
  >
    <label
      className="checkbox-container txt-s block mb6 flex"
      key="Option value 1"
    >
      <input
        aria-checked={false}
        autoFocus={false}
        checked={false}
        name="testinput"
        onChange={[Function]}
        role="checkbox"
        type="checkbox"
        value="Option value 1"
      />
      <div
        className="mr6 inline-block checkbox--blue checkbox--s-label checkbox"
      >
        <Icon
          name="check"
        />
      </div>
      Option 1
    </label>
    <label
      className="checkbox-container txt-s block mb6 flex"
      key="Option value 2"
    >
      <input
        aria-checked={false}
        autoFocus={false}
        checked={false}
        disabled={true}
        name="testinput"
        onChange={[Function]}
        role="checkbox"
        type="checkbox"
        value="Option value 2"
      />
      <div
        className="mr6 inline-block checkbox--blue checkbox--s-label checkbox"
      >
        <Icon
          name="check"
        />
      </div>
      Option 2
    </label>
  </div>
</ControlWrapper>
`;
